src/devtools/useAtomsDevtools.ts

```diff
@@ -0,0 +1,174 @@
+  error: (payload: any) => void
+}
+
+type Extension = {
+  connect: (options?: Config) => ConnectionResult
+}
+
+type AtomsSnapshot = Map<Atom<unknown>, unknown>
+
+const serializeSnapshot = (snapshot: AtomsSnapshot) => {
+  const result: Record<string, unknown> = {}
+
+  snapshot.forEach((v, atom) => {
+    result[atomToDebuggingString(atom)] = v
+  })
+
+  return result
+}
+
+const atomToDebuggingString = (atom: Atom<unknown>) =>
+  `${atom}:${atom.debugLabel ? atom.debugLabel : atom}`
+
+const getDependencies = (store: Store, snapshot: AtomsSnapshot) => {
+  const result: Record<string, string[]> = {}
+
+  snapshot.forEach((_, atom) => {
+    const atomState = store[DEV_GET_ATOM_STATE]?.(atom)
+
+    if (!atomState) {
+      return
+    }
+    result[atomToDebuggingString(atom)] = [...atomState.d.keys()].map(
+      atomToDebuggingString
+    )
+  })
+
+  return result
+}
+
+export function useAtomsDevtools(name: string, scope?: Scope) {
+  const snapshot = useAtomsSnapshot()
+  const goToSnapshot = useGotoAtomsSnapshot()
+  let extension: Extension | undefined
+  try {
+    extension = (window as any).__REDUX_DEVTOOLS_EXTENSION__ as Extension
+  } catch {}
+  if (!extension) {
+    if (
+      typeof process === 'object' &&
+      process.env.NODE_ENV === 'development' &&
+      typeof window !== 'undefined'
+    ) {
+      console.warn('Please install/enable Redux devtools extension')
+    }
+  }
+
+  const ScopeContext = getScopeContext(scope)
+  const scopeContainer = useContext(ScopeContext)
+  const store = scopeContainer.s
+
+  if (!store[DEV_SUBSCRIBE_STATE]) {
+    throw new Error('useAtomsSnapshot can only be used in dev mode.')
+  }
+
+  const isTimeTraveling = useRef(false)
+  const isRecording = useRef(true)
+  const devtools = useRef<ConnectionResult & { shouldInit?: boolean }>()
+  const snapshots = useRef<AtomsSnapshot[]>([])
+
+  useEffect(() => {
+    let devtoolsUnsubscribe: () => void | undefined
+    if (extension) {
+      devtools.current = extension.connect({ name })
+
+      devtoolsUnsubscribe = devtools.current.subscribe((message: Message) => {
+        switch (message.type) {
+          case 'ACTION':
+            return
+          case 'DISPATCH':
+            switch (message.payload?.type) {
+              case 'RESET':
+                // Todo
+                return
+              case 'COMMIT':
+                devtools.current!.init(undefined)
+                return
+              case 'JUMP_TO_STATE':
+              case 'JUMP_TO_ACTION':
+                isTimeTraveling.current = true
+
+                const snapshot =
+                  snapshots.current[message.payload.actionId - 1]!
+
+                goToSnapshot(snapshot)
+                return
+              case 'PAUSE_RECORDING':
+                return (isRecording.current = !isRecording.current)
+            }
+        }
+      })
+
+      devtools.current.shouldInit = true
+    }
+    return () => {
+      devtoolsUnsubscribe?.()
+    }
```

Comment:
```suggestion
    return devtoolsUnsubscribe
```
can work?

Reply:
hmm, but doesn't work well with types

Reply:
Oh, this is not correct:
```ts
let devtoolsUnsubscribe: () => void | undefined
```
It has to be:
```ts
let devtoolsUnsubscribe: (() => void) | undefined
```